portalnet/src/overlay_service.rs

```diff
@@ -1588,7 +1588,7 @@ where
             }
 
             // close uTP connection
-            if let Err(err) = stream.shutdown() {
+            if let Err(err) = stream.close().await {
```

Comment:
IIUC, `close()` now confirms that all of the data was sent correctly? (Rather than just pushed into the send buffer which was the behavior for `shutdown()`). If so, it looks like we can remove some of the logic inside [`handle_find_content()`](https://github.com/ethereum/trin/blob/master/portalnet/src/overlay_service.rs#L1110-L1149) where we check the `write_size` to validate that a utp tx was successful.

Reply:
Oh yeah, this called for a wider refactor: the duplicate code in handle_find_content had already drifted significantly. So I combined the two usages into a single function call. This also helped us get rid of the tracking boolean flag for whether we had recorded the utp outcome.